devenv.md

```diff
@@ -0,0 +1,345 @@
+# Setup your LeApp development environment
+
+  Currently, LeApp tool is divided into multiple projects. This means that in order to setup
+a development environment of the LeApp project one has to setup everything piece by piece.
+At this point there are basically two choices, either setup minimal viable environment which
+includes [*Snactor library*](#snactor), [*LeApp daemon*](#daemon) and [*LeApp tool*](#leapp)
+which includes Cockpit UI plugin, or setup a full fledged development environment which contains
+all the bits and pieces mentioned in section [*2*](#repositories). The choice of which kind of
+environment should be setup depends on the needs. However, the recommended one is the full fledged
+environment.
+  In order to start the setup procedure, please follow all steps in section [1 - Prerequisites](#prereq)
+since every step in the following section [*2*](#repositories) depends on it. Then you can proceed and
+set up each required project. This should be again straigt forward process, as it requires following
+each step mentioned in the project section.
+
+  Because we realize that this is quite complicated and rather tedious process, we prepared a [script](#bscript)
+and a set of pre-prepared [images](#vagrant).
+
+## <a name="prereq"></a>1. Prerequisites
+
+**Step 1) Enabling repositories containing build & runtime libraries:**
+
+```shell
+    sudo dnf copr enable evilissimo/leapp
+```
+
+**Step 2) Installing required tools:**
+
+```shell
+    sudo dnf -y install git tito python2-pip
+```
+
+**Step 3) Setting up a project root**
+```shell
+    export LEAPP_HOME=$HOME/leapp
+    mkdir -p $LEAPP_HOME
+```
+
+
+## <a name="repositories"></a>2. LeApp repositories
+
+### <a name="snactor"></a>2.1 Snactor library
+Snactor library is a runtime dependency of LeApp daemon. This library is responsible for
+executing and handling of so-called actors.
+
+#### Step 1) Repository cloning \*
+
+```shell
+    git clone https://github.com/leapp-to/snactor.git $LEAPP_HOME/snactor
+```
+
+\* *Fork the repository if you want to open pull requests in the future*
+
+#### Step 2) External dependencies & packages
+
+```shell
+    sudo dnf -y install ansible
+```
+
+#### Step 3) Build & installation:
+
+```shell
+    cd $LEAPP_HOME/snactor
+    sudo dnf builddep python-snactor.spec
+    tito build --install --verbose --rpm --test
+```
+
+#### Step 4) Setting up a testing environment - unit testing:
+
+```shell
+    cd $LEAPP_HOME/snactor
+    pip install --user -r requirements-tests.txt
+
+    sudo dnf -y install tox libffi libffi-devel openssl openssl-devel
```

Comment:
IMO `tox` should be added to `requirements-test.txt` instead of installing via `dnf`.

I'm not sure if we need the other packages, though. Are they dependencies of some of snactor dependencies?